### PR TITLE
Xufang/enable cc for graph optimizer

### PR DESCRIPTION
**Pull request by xufang-lisa**
### Details:
 - *add cc for graph optimizer*

### Tickets:
 - *102652*

### CODE REVIEW

File: src/plugins/intel_cpu/src/itt.h [hunk 1]
```diff
@@ -27,10 +27,15 @@ OV_ITT_DOMAIN(intel_cpu_LT);
 
 #if defined(SELECTIVE_BUILD_ANALYZER)
 #    define CPU_LPT_SCOPE(region) OV_SCOPE(intel_cpu, region)
+#    define CPU_GRAPH_OPTIMIZER_SCOPE(region) OV_SCOPE(intel_cpu, region)
 #elif defined(SELECTIVE_BUILD)
 #    define CPU_LPT_SCOPE(region)                                          \
         if (OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(intel_cpu, _, region)) == 0) \
         throw ov::Exception(std::string(OV_PP_TOSTRING(OV_PP_CAT3(ov_op, _, region))) + " is disabled!")
+#    define CPU_GRAPH_OPTIMIZER_SCOPE(region)                              \
+        if (OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(intel_cpu, _, region)) == 0) \
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/itt.h:36]: Does it work? It seems that here is only a scope check?

Comment (riverlijunjie) [reply src/plugins/intel_cpu/src/itt.h:36]: Do you have some test data?

**Comment by dmitry-gorokhov** [comment]
@xufang-lisa  @riverlijunjie  guys, could you please provide any estimation on binary size reduction with this optimization?

**Comment by riverlijunjie** [comment]
> @xufang-lisa @riverlijunjie guys, could you please provide any estimation on binary size reduction with this optimization?

yes, there are about 70KB binary size reduction on Linux static build, and WIP to collect the data on Windows.

**Comment by xufang-lisa** [comment]
> > @xufang-lisa @riverlijunjie guys, could you please provide any estimation on binary size reduction with this optimization?
> 
> yes, there are about 70KB binary size reduction on Linux static build, and WIP to collect the data on Windows.

on windows static build, there are about 50KB binary size reduction.

**Review by riverlijunjie — APPROVED** [review]
Binary size benefit is ~50KB for windows and ~90KB for Linux, it is pretty promising.

**Comment by ilyachur** [comment]
@dmitry-gorokhov Please take a look.

@xufang-lisa Can we also reuse `ADD_MATCHER` inside the CPU plugin?

**Comment by xufang-lisa** [comment]
> @dmitry-gorokhov Please take a look.
> 
> @xufang-lisa Can we also reuse `ADD_MATCHER` inside the CPU plugin?

I think that 'ADD_MATCHER' can not be reused for this graph optimizer, 'ADD_MATCHER' is defined as whether to perform add_matcher operation.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**